cheroot/ssl/pyopenssl.py

```diff
@@ -328,13 +333,28 @@ def wrap(self, sock):
         # closing so we can't reliably access protocol/client cert for the env
         return sock, self._environ.copy()
 
+    def _password_callback(
+        self,
+        _password_max_length,
+        _verify_twice,
+        password,
+    ):
+        """Pass a passphrase to password protected private key."""
+        if not password:
+            return b''
+        if not isinstance(password, bytes):
+            return password.encode('utf-8')
+        return password
+
     def get_context(self):
         """Return an ``SSL.Context`` from self attributes.
 
         Ref: :py:class:`SSL.Context <pyopenssl:OpenSSL.SSL.Context>`
         """
         # See https://code.activestate.com/recipes/442473/
         c = SSL.Context(SSL.SSLv23_METHOD)
+        if self.private_key_password is not None:
+            c.set_passwd_cb(self._password_callback, self.private_key_password)
```

Comment:
https://www.pyopenssl.org/en/stable/api/ssl.html#OpenSSL.SSL.Context.set_passwd_cb says
> Set the passphrase callback. This function will be called when a private key with a passphrase is loaded.

This probably means that it can be called unconditionally and pyOpenSSL will ignore it when the private key is non-encrypted, right?
```suggestion
        c.set_passwd_cb(self._password_callback, self.private_key_password)
```

Reply:
Yes, it seems so. Modified the code like yo suggested and I added unit test for this use case.